### ISSUE TITLE
Execute Async?

### ISSUE THREAD

**Issue by Seabizkit** [SteveSandersonMS/DotNetIsolator#24]
how do i execute  async function

![image](https://github.com/SteveSandersonMS/DotNetIsolator/assets/1920090/f653408a-0532-49c5-8eb2-d9c531b690f2)

cant wait for the day when this stuff is simple.... my god is it hard to say... here is some code run in isolation.

**Comment by moshe-saccal** [comment]
I would say , since wasm is single threaded, there's no reason to run async code...
am i wrong?

**Comment by Seabizkit** [comment]
> I would say , since wasm is single threaded, there's no reason to run async code... am i wrong?

yes, wasm is a sandbox for hardware resources ... so not only would u want access to mutiple threads but also async capabilities

**Comment by Seabizkit** [comment]
is this project still being maintained../updated .... i personal think this is amazing project i just want it to be capitiable of more

**Comment by cecilphillip** [comment]
I believe at the moment this is just an experiment / PoC. I wouldn't expect too much in terms of "maintenance".

**Comment by Shadowblitz16** [comment]
> I would say , since wasm is single threaded, there's no reason to run async code... am i wrong?

I think the issue is running multiple wasi instances asynchronously on the host not multiple threads in the wasi instance.

I could be wrong though.
